### PR TITLE
properly handle emtpy selectors in withEach

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/jenkins-client-plugin/issues/131

@openshift/sig-developer-experience ptal

### CODE REVIEW

**Comment by gabemontero** [comment]
bump @coreydaley @bparees - any bandwidth / interest?

**Comment by bparees** [comment]
no :(

File: src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy [hunk 1]
```diff
@@ -1519,8 +1519,10 @@ class OpenShiftDSL implements Serializable {
         public <V> V withEach(Closure<V> body) {
             defaultProjectsForObjectList();
             List<String> names = names();
+            if (names == null || names.size() == 0)
+                return;
             List<String> projects = projects();
-            if (projects == null) {
+            if (projects == null || projects.size != names.size()) {
```
Comment (coreydaley) [inline src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy]: should this be projects.size() ?

Comment (gabemontero) [reply src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy]: ahh ... good catch ... not so safe after all :-)

Comment (gabemontero) [reply src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy]: and another instance of stupid groovy ;-)

Comment (coreydaley) [reply src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy]: Is there any kind of groovy linter that we could add to the testing for these repos?

**Comment by gabemontero** [comment]
On Thu, Mar 22, 2018 at 10:51 AM, Corey Daley <notifications@github.com>
wrote:

> *@coreydaley* commented on this pull request.
> ------------------------------
>
> In src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy
> <https://github.com/openshift/jenkins-client-plugin/pull/132#discussion_r176449000>
> :
>
> >              List<String> projects = projects();
> -            if (projects == null) {
> +            if (projects == null || projects.size != names.size()) {
>
> Is there any kind of groovy linter that we could add to the testing for
> these repos?
>
Yeah there appears to be some, both for groovy in general and jenkins
pipelines.
A simple "groovy linter" search turns up a bunch of stuff

I just have not had the time to try to wedge them into or repo mechanics.
Certainly
feel free to hack away if you think you got the time.

Also, several IDEs are suppose to have linters for groovy.  Whether it it
switching
to them, or finally upgrading my eclipse in the hopes I get better groovy
support,
are on my list too.

I also want to move the vast majority of our logic out of
OpenShiftDSL.groovy into
a java file, and just use the groovy file as a bridge.  Even have a card
open for that one.

**Pull request merged**